gdal/ogr/ogrsf_frmts/flatgeobuf/ogrflatgeobuflayer.cpp

```diff
@@ -239,11 +253,27 @@ void OGRFlatGeobufLayer::readColumns()
         return;
     for (uint32_t i = 0; i < columns->size(); i++) {
         const auto column = columns->Get(i);
+        const auto type = column->type();
         const auto name = column->name()->c_str();
+        const auto title = column->title() != nullptr ? column->title()->c_str() : nullptr;
+        const auto width = column->width();
+        const auto precision = column->precision();
+        const auto scale = column->scale();
+        const auto nullable = column->nullable();
+        const auto unique = column->unique();
         OGRFieldSubType eSubType = OFSTNone;
-        const auto type = toOGRFieldType(column->type(), eSubType);
-        OGRFieldDefn field(name, type);
+        const auto ogrType = toOGRFieldType(column->type(), eSubType);
+        OGRFieldDefn field(name, ogrType);
         field.SetSubType(eSubType);
+        field.SetAlternativeName(title);
+        if (width != 1 && type != ColumnType::Float && type != ColumnType::Double)
```

Comment:
there's a comparison to 1 against width, and below against -1 for precision. Intended ?

Reply:
Good catch, it was a typo.